### PR TITLE
Make isSameDimensions optional [Fixes #313]

### LINKED ISSUE #313: Components failing tests when mismatch percent is near zero.
Our mismatchThreshold is set as follows: `"misMatchThreshold": 5,` however when the tests run certain tests fail and say `"isSameDimensions": false,`.

Here's an example of some screenshots where the tests _should_ pass but don't seem to.

prod_test_hillarys_styleguide_30_introbanner_2_desktop

prod_test_hillarys_styleguide_30_introbanner_2_desktop 1

Here's the output result from the report:

```
Report: {
  "isSameDimensions": false,
  "dimensionDifference": {
    "width": 0,
    "height": -1
  },
  "misMatchPercentage": "0.48"
}
Threshold: 1
```

This happens on around 8-10 tests, intermittently. Are we missing something, or is there a way to turn off the `isSameDimensions` property? Is this expected?

Thanks in advance for your advice.

### PR DESCRIPTION
**Pull request by jtuds**

### CODE REVIEW

**Comment by jtuds** [comment]
The bump in versions done yesterday should fix this CI build

File: capture/config.default.json [hunk 1]
```diff
@@ -37,6 +37,7 @@
       "readyEvent": null,
       "delay": 500,
       "misMatchThreshold" : 0.1,
+      "isSameDimensions" : true,
```
Comment (garris) [inline capture/config.default.json]: Would you consider changing property name to `requireSameDimentions`?

File: core/util/makeConfig.js [hunk 2]
```diff
@@ -100,6 +100,7 @@ function makeConfig (argv) {
 
   config.defaultMisMatchThreshold = 0.1;
   config.debug = userConfig.debug || false;
+  config.defaultIsSameDimensions = userConfig.isSameDimensions || true;
```
Comment (garris) [inline core/util/makeConfig.js]: This will always resolve to true.   Try `userConfig.isSameDimensions === false ? false : true`

**Comment by garris** [comment]
Thank you for contributing.   A lot of developers have requested this -- please take a look at the comments and follow up.  Cheers!

**Comment by cheapaschips** [comment]
Can't wait for this addition to be merged, I'm currently stuck with my project until this has the green light due to soooo many failures due to the height being flagged.

**Comment by jtuds** [comment]
I'm going to work on the requested changes soon. Hopefully get this merged in soon after that :)

**Comment by garris** [comment]
@jtuds thanks!

@cheapaschips just curious, are you able to just remove the elements that are causing the issue?  Or use content stubs so that you will know the correct height in advance?

**Comment by cheapaschips** [comment]
@garris I'm not entirely sure which element is causing the height to change. Looking at the diff it's as if the initial screenshot is sharp and the test screenshot is fuzzy which sometime moves the height randomly from -1 to -22. I'm not able to use content stubs as we need a tool to run in the production environment.

**Comment by cheapaschips** [comment]
Hey @jtuds @garris have you managed to look into this fix by any chance?

Many thanks

**Comment by jtuds** [comment]
@cheapaschips I haven't yet had a chance to properly implement the changes. It's on my list to do, perhaps might get chance tomorrow!

**Comment by cheapaschips** [comment]
@jtuds that's would be amazing!

**Comment by jtuds** [comment]
Requested changes done and pull request updated

**Comment by stevecfischer** [comment]
@jtuds - I'm reviewing your pull request. I get the error 
```
The package eslint-plugin-promise@2.0.1 does not satisfy its siblings' peerDependencies requirements!
npm ERR! peerinvalid Peer eslint-config-standard@6.2.1 wants eslint-plugin-promise@>=3.3.0
```
I think you just need to pull from master to fix it?!?

**Comment by jtuds** [comment]
Sorry about that. I think the dependencies have changed since I did the work. I've merged master in to the branch. @stevecfischer

**Comment by jtuds** [comment]
Master merged again after conflicts found

File: capture/config.default.json (hunk 1 above)
Comment (cdaringe) [inline capture/config.default.json]: i find `requireSameDimentions` more clear as well.  thx!

File: capture/genBitmaps.js [hunk 3]
```diff
@@ -231,6 +231,7 @@ function processScenario (casper, scenario, scenarioOrVariantLabel, scenarioLabe
             selector: o,
             fileName: fileName,
             label: scenario.label,
+            isSameDimensions: scenario.isSameDimensions || config.isSameDimensions || config.defaultIsSameDimensions,
```
Comment (cdaringe) [inline capture/genBitmaps.js]: do these need to be checking for falsly values? for example, if the scenario says false, but the config says true, the higher precedence scenario will be trumped

**Comment by cdaringe** [comment]
@jtuds, if you need help finishing up the addition change requests, let me know.

**Comment by jtuds** [comment]
@cdaringe I'll hopefully get round to it sooner rather than later (I don't know when!) but anyone is more than welcome to help push this over the line sooner :)

**Comment by cdaringe** [comment]
pulling this now to finish the work

**Comment by cdaringe** [comment]
i pulled it, only to find that it looks like the follow ups were already addressed. are we g2g on this?

**Comment by jtuds** [comment]
@cdaringe As far as I'm aware, the issue you raised in capture/genBitmaps.js has not been fixed. Check for falsy values needs to be implemented. Thanks for helping by the way!

**Comment by garris** [comment]
@jtuds thank you very much for your patience on this one!!!   It's on canary.  I hand merged this with a performance upgrade.  Hopefully I didn't screw anything up.  

`npm install backstopjs@canary -g`

Please let me know if everything is working ok.

Thanks again for all your help!

**Pull request merged**